tensorflow/core/framework/node_def.proto

```diff
@@ -35,7 +35,7 @@ message NodeDef {
   // CONSTRAINT ::= ("job:" JOB_NAME)
   //              | ("replica:" [1-9][0-9]*)
   //              | ("task:" [1-9][0-9]*)
-  //              | ( ("gpu" | "cpu") ":" ([1-9][0-9]* | "*") )
+  //              | ("device:" ("gpu" | "cpu") ":" ([1-9][0-9]* | "*") )
```

Comment:
Technically it can be /device:(^:)* (anything but ":"), but probably we can just say A-Za-z to keep things simple.

Reply:
Ack'd, sorry for delay vrv@.  We can correct this with another PR if you'd like